src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,141 @@
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../TodoContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, completed, title } = todo;
+
+  const [isEdding, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEdding && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEdding]);
+
+  const {
+    deleteTodo,
+    isDisabled,
+    tempTodoIds,
+    updateTodo,
+  } = useContext(TodosContext);
+
+  const handleToggleChange = () => {
+    const newTodo = ({ ...todo, completed: !completed });
+
+    updateTodo(newTodo);
+  };
+
+  const handleOnEscape = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Escape') {
+      setIsEditing(false);
+      setNewTitle(title);
+    }
+  };
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+  };
+
+  const aditFunc = (str: string) => {
+    const trimmedTitle = str.trim();
+
+    if (!trimmedTitle) {
+      deleteTodo(id);
+      setIsEditing(false);
+      setNewTitle(title);
+
+      return;
+    }
+
+    const todoUpdated = { ...todo, title: trimmedTitle };
+
+    updateTodo(todoUpdated);
+
+    setIsEditing(false);
+  };
+
+  const handleOnBlur = () => {
+    aditFunc(newTitle);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    aditFunc(newTitle);
+  };
+
+  return (
+    <div data-cy="Todo" className={cn('todo', { completed })}>
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={handleToggleChange}
+        />
+      </label>
+
+      {isEdding ? (
+        <form onSubmit={handleSubmit}>
+          <input
+            ref={inputRef}
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={newTitle}
+            onChange={handleOnChange}
+            onKeyUp={handleOnEscape}
+            onBlur={handleOnBlur}
+          />
+        </form>
+      ) : (
+        <>
+          <span
+            data-cy="TodoTitle"
+            className="todo__title"
+            onDoubleClick={() => setIsEditing(true)}
+          >
+            {newTitle || title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => deleteTodo(id)}
+          >
+            ×
+          </button>
+        </>
+
+      )}
+
+      {/* overlay will cover the todo while it is being updated */}
```

Comment:
Remove comment  
Try to write obvious code with meaningful variable and function naming